pkg/types/powervs/platform.go

```diff
@@ -0,0 +1,63 @@
+package powervs
+
+// Platform stores all the global configuration that all machinesets
+// use.
+// Note: The subsequent mentions of future-TF support refer to work that is
+// undergoing and should be available to test well in time for 4.10 feature-
+// freeze. We do not plan to GA with these as required inputs.
+type Platform struct {
+
+	// ServiceInstanceID is the ID of the Power IAAS instance created from the IBM Cloud Catalog
+	ServiceInstanceID string `json:"serviceInstance"`
+
+	// PowerVSResourceGroup is the resource group for creating Power VS resources.
+	PowerVSResourceGroup string `json:"powervsResourceGroup"`
+
+	// Region specifies the IBM Cloud region where the cluster will be created.
+	Region string `json:"region"`
+
+	// Zone specifies the IBM Cloud colo region where the cluster will be created.
+	// Required for multi-zone regions.
+	Zone string `json:"zone"`
+
+	// Zone in the region used to create VPC resources. Leave unset
+	// to allow installer to randomly select a zone.
+	//
+	// +optional
+	VPCZone string `json:"vpcRegion,omitempty"`
+
+	// UserID is the login for the user's IBM Cloud account.
+	UserID string `json:"userID"`
+
+	// APIKey is the API key for the user's IBM Cloud account.
+	APIKey string `json:"APIKey,omitempty"`
```

Comment:
These should not be in the install-config.yaml. Can these be read from another file instead? Or from the IC_API_KEY environment variable as with the ibmcloud platform?

Reply:
yah, we can remove that. @cklokman would you mind working on this?

Reply:
n/m @cklokman